numpy/__init__.pyi

```diff
@@ -253,15 +253,121 @@ class _ArrayOrScalarCommon(
+    def tobytes(self, order: _OrderKACF = ...) -> bytes: ...
+    def tofile(
+        self, fid: Union[IO[bytes], str], sep: str = ..., format: str = ...
+    ) -> None: ...
+    # generics and 0d arrays return builtin scalars
+    def tolist(self) -> Any: ...
+    @overload
+    def transpose(self: _ArraySelf, axes: Sequence[int]) -> _ArraySelf: ...
+    @overload
+    def transpose(self: _ArraySelf, *axes: int) -> _ArraySelf: ...
+    @overload
+    def view(self, type: Type[_NdArraySubClass]) -> _NdArraySubClass: ...
+    @overload
+    def view(self: _ArraySelf, dtype: DtypeLike = ...) -> _ArraySelf: ...
+    @overload
+    def view(
+        self, dtype: DtypeLike, type: Type[_NdArraySubClass]
+    ) -> _NdArraySubClass: ...
+
+    # TODO: Add proper signatures
+    def __getitem__(self, key) -> Any: ...
+    @property
+    def __array_interface__(self): ...
+    @property
+    def __array_priority__(self): ...
+    @property
+    def __array_struct__(self): ...
+    def __array_wrap__(array, context=...): ...
+    def __setstate__(self, __state): ...
+    def all(self, axis=..., out=..., keepdims=...): ...
+    def any(self, axis=..., out=..., keepdims=...): ...
+    def argmax(self, axis=..., out=...): ...
+    def argmin(self, axis=..., out=...): ...
+    def argpartition(self, kth, axis=..., kind=..., order=...): ...
+    def argsort(self, axis=..., kind=..., order=...): ...
+    def choose(self, choices, out=..., mode=...): ...
+    def clip(self, min=..., max=..., out=..., **kwargs): ...
+    def compress(self, condition, axis=..., out=...): ...
+    def conj(self): ...
+    def conjugate(self): ...
+    def cumprod(self, axis=..., dtype=..., out=...): ...
+    def cumsum(self, axis=..., dtype=..., out=...): ...
+    def diagonal(self, offset=..., axis1=..., axis2=...): ...
+    def dot(self, b, out=...): ...
+    def max(self, axis=..., out=..., keepdims=..., initial=..., where=...): ...
+    def mean(self, axis=..., dtype=..., out=..., keepdims=...): ...
+    def min(self, axis=..., out=..., keepdims=..., initial=..., where=...): ...
+    def newbyteorder(self, new_order=...): ...
+    def nonzero(self): ...
+    def partition(self, kth, axis=..., kind=..., order=...): ...
+    def prod(self, axis=..., dtype=..., out=..., keepdims=..., initial=..., where=...): ...
+    def ptp(self, axis=..., out=..., keepdims=...): ...
+    def put(self, indices, values, mode=...): ...
+    def repeat(self, repeats, axis=...): ...
+    def round(self, decimals=..., out=...): ...
+    def searchsorted(self, v, side=..., sorter=...): ...
+    def setfield(self, val, dtype, offset=...): ...
+    def sort(self, axis=..., kind=..., order=...): ...
+    def std(self, axis=..., dtype=..., out=..., ddof=..., keepdims=...): ...
+    def sum(self, axis=..., dtype=..., out=..., keepdims=..., initial=..., where=...): ...
+    def take(self, indices, axis=..., out=..., mode=...): ...
+    def tostring(self, order=...): ...
```

Comment:
If it's relevant, `tostring` is deprecated to we could perhaps skip it

Reply:
Yeah we have in fact removed it from the subs before...

Reply:
I'll add a note in a bit as a future reminder.

Reply:
And it's been removed (again) as of https://github.com/numpy/numpy/pull/17188/commits/5ea2ec96779cb529776487b26a261ad08a43c74a.